src/parse.y

```diff
@@ -206,8 +206,14 @@ option		:  TOK_OUTFILE '=' NAME
 			{ yyclass = xstrdup(nmstr); }
 		|  TOK_HEADER_FILE '=' NAME
 			{ headerfilename = xstrdup(nmstr); }
-	    |  TOK_TABLES_FILE '=' NAME
-            { tablesext = true; tablesfilename = xstrdup(nmstr); }
```

Comment:
@whydoubt 
I apologize first for overlooking the "command-line argument" part of the code.
I found it in main.c:
```
		case OPT_TABLES_FILE:
			tablesext = true;
			tablesfilename = arg;
			break;
```
But still, having an additional `is_tablesfilename_alloc` doesn't convince me.
I'll suggest another way:
First, change the aforementioned code in main.c to `tablesfilename = xstrdup(arg);` that is, always strdup the name string before assigning to the global `tablesfilename` variable.
Then, for `%option tables-file` or additional `--tables-file` in the argument, check if `tablesfilename` has been assigned before. If yes, we can print a warning ("foo.l: warning: multiple specification of %option tables-file; ignoring all but last one"), and then you can do your memory leak fixes.